src/libraries/System.Linq.Queryable/src/System/Linq/EnumerableRewriter.cs

```diff
@@ -255,25 +255,83 @@ protected override Expression VisitConstant(ConstantExpression c)
         }
 
         private static ILookup<string, MethodInfo>? s_seqMethods;
-        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2060:MakeGenericMethod",
-            Justification = "Enumerable methods don't have trim annotations.")]
         private static MethodInfo FindEnumerableMethodForQueryable(string name, ReadOnlyCollection<Expression> args, params Type[]? typeArgs)
         {
-            if (s_seqMethods == null)
+            s_seqMethods ??= GetEnumerableStaticMethods(typeof(Enumerable)).ToLookup(m => m.Name);
+
+            MethodInfo[] matchingMethods = s_seqMethods[name]
+                .Where(m => ArgsMatch(m, args, typeArgs))
```

Comment:
Is it possible to fix the issue in `ArgsMatch`? It seems odd that it would return `true` when the method takes a `TSource` as a parameter, and the `args` has an `Expression<Func<TSource, bool>>`.

Reply:
I don't think the `ArgsMatch` method is doing anything incorrectly, this is an issue of picking the correct matching overload.